File: src/widget.ts
// Copyright (c) {{ cookiecutter.author_name }}
// Distributed under the terms of the Modified BSD License.

import {
  DOMWidgetModel,
  DOMWidgetView
} from '@jupyter-widgets/base';
import type {  ISerializers} from '@jupyter-widgets/base';
import { setStoreModels } from './stores';

import { MODULE_NAME, MODULE_VERSION } from './version';

import Widget from './components/Widget.svelte'

export class SetMLVisModel extends DOMWidgetModel {
  //These are the defaults used when excuting the model
  // Any variables add to widget.py should also be added here
  defaults() {
    return {
      ...super.defaults(),
      _model_name: SetMLVisModel.model_name,
      _model_module: SetMLVisModel.model_module,
      _model_module_version: SetMLVisModel.model_module_version,
      _view_name: SetMLVisModel.view_name,
      _view_module: SetMLVisModel.view_module,
      _view_module_version: SetMLVisModel.view_module_version,
      data: {},
      num_instance: 0,
      height: 800,
      IOU:0.8,
    };
  }

  static serializers: ISerializers = {
    ...DOMWidgetModel.serializers,
    // Add any extra serializers here
  };

  static model_name = 'SetMLVisModel';
  static model_module = MODULE_NAME;
  static model_module_version = MODULE_VERSION;
  static view_name = 'SetMLVisView'; // Set to null if no view
  static view_module = MODULE_NAME; // Set to null if no view
  static view_module_version = MODULE_VERSION;
}

export class SetMLVisView extends DOMWidgetView {
  render() {
    setStoreModels(this.model);
    new Widget({ target: this.el });
  }
}

File: setmlvis/widget.py
#!/usr/bin/env python
# coding: utf-8

# Copyright (c) {{ cookiecutter.author_name }}.
# Distributed under the terms of the Modified BSD License.

"""
TODO: Add module docstring
"""

from ipywidgets import DOMWidget
from traitlets import Unicode, Int, Float, Dict
from typing import Union
from pathlib import Path
import json
from ._frontend import module_name, module_version


class SetMLVisWidget(DOMWidget):
    """TODO: Add docstring here
    """
    _model_name = Unicode('SetMLVisModel').tag(sync=True)
    _model_module = Unicode(module_name).tag(sync=True)
    _model_module_version = Unicode(module_version).tag(sync=True)
    _view_name = Unicode('SetMLVisView').tag(sync=True)
    _view_module = Unicode(module_name).tag(sync=True)
    _view_module_version = Unicode(module_version).tag(sync=True)

    dataset = Dict({}).tag(sync=True)

    num_instances = Int(0).tag(sync=True)
    IOU = Float(0.8).tag(sync=True)
    height = Int(800).tag(sync=True)
    # selectedCol = List([]).tag(sync=True)
    selectedImgs = Dict({}).tag(sync=True)
    def __init__(
        self,
        data: Union[str, Path, dict],
        height: int = 800,
        IOU: float = 0.8,
        **kwargs,
    ):
        super().__init__(**kwargs)

        # if data is a path or string, then read the file at that path
        # if isinstance(data, Path) or isinstance(data, str):
        #     path = Path(data).resolve()

        #     if not path.exists():
        #         raise OSError(f"Cannot read {path}")

        #     json_data = path.read_text(encoding="utf-8")
        #     data = json.loads(json_data)


        # synced widget state
        self.dataset = data
        self.num_instances = 10
        self.height = height
        self.IOU = IOU